analysis/dataset_definition_dm017.py

```diff
@@ -7,8 +8,10 @@
     get_dm_reg_r2,
 )
 
-# Define index date
-index_date = date(2022, 3, 31)
+parser = ArgumentParser()
+parser.add_argument("--index-date", type=str)
```

Comment:
You can avoid having to parse the date separately by doing:
```py
parser.add_argument("--index-date", type=datetime.date.fromisoformat)
```

The `type` argument just needs to be a callable which takes a string and returns a value of the type you want.

Note that if you do this, and you want to supply a default, then the default needs to match the argument type rather than being a string. For example:
```py
parser.add_argument(
    "--index-date",
    type=datetime.date.fromisoformat,
    default=datetime.date(2020, 1, 1),
)